From cd42b475399c84767d2cc2940ce7964c07a2bac8 Mon Sep 17 00:00:00 2001
From: Ari Parkkila <ari.parkkila@qt.io>
Date: Thu, 20 Oct 2022 08:37:47 +0300
Subject: [PATCH] Fix API break in init_disassemble_info

API break due to init_disassemble_info() got an extra parameter,
see http://trofi.github.io/posts/254-binutils-2.39.html

Upstream-Status: Backport
---
 tools/perf/util/annotate.c | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/tools/perf/util/annotate.c b/tools/perf/util/annotate.c
index e42bf572358c..779e9d5fdb16 100644
--- a/tools/perf/util/annotate.c
+++ b/tools/perf/util/annotate.c
@@ -1708,6 +1708,18 @@ static int dso__disassemble_filename(struct dso *dso, char *filename, size_t fil
 #include <bfd.h>
 #include <dis-asm.h>
 
+static int fprintf_styled(void *, enum disassembler_style, const char* fmt, ...)
+{
+	va_list args;
+	int r;
+
+	va_start(args, fmt);
+	r = vprintf(fmt, args);
+	va_end(args);
+
+	return r;
+}
+
 static int symbol__disassemble_bpf(struct symbol *sym,
 				   struct annotate_args *args)
 {
@@ -1750,7 +1762,8 @@ static int symbol__disassemble_bpf(struct symbol *sym,
 		goto out;
 	}
 	init_disassemble_info(&info, s,
-			      (fprintf_ftype) fprintf);
+			      (fprintf_ftype) fprintf,
+			      fprintf_styled);
 
 	info.arch = bfd_get_arch(bfdf);
 	info.mach = bfd_get_mach(bfdf);
-- 
2.25.1

